### PR TITLE
Use summation model in app

### PR DESCRIPTION
**Pull request by dfulu**
# Pull Request

## Description

- Add the summation model to the app
- Update tests

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet/app.py [hunk 1]
```diff
@@ -234,6 +248,9 @@ def app(
         .effective_capacity_mwp
     )
 
+    # National capacity is needed if using summation model
+    national_capacity = gsp_capacities.sum().item()
```
Comment (peterdudfield) [inline pvnet/app.py:252]: should we take the sum here or should we take the national one from the database?

File: pvnet/app.py [hunk 2]
```diff
@@ -284,15 +301,36 @@ def app(
...
+        ):
+            warnings.warn(
+                f"The PVNet version running in this app is "
+                f"{model_name}/{model_version}."
+                f"The summation model running in this app was trained on outputs from PVNet "
+                f"version {summation_model.model_name}/{summation_model.model_version}. "
+                f"Combining these models may lead to an error if the shape of PVNet output doesn't "
+                f"match the expected shape of the summation model. Combining may lead to "
+                f"unreliable results even if the shapes match."
+            )
```
Comment (peterdudfield) [inline pvnet/app.py:327]: nice

File: pvnet/app.py [hunk 3]
```diff
@@ -313,19 +351,27 @@ def app(
             elevation = elevation[:, -preds.shape[1] :]
             sun_down_mask = elevation < MIN_DAY_ELEVATION
 
-            # Zero out after sundown
-            preds[sun_down_mask] = 0
```
Comment (peterdudfield) [inline pvnet/app.py:317]: why remove this?

Comment (peterdudfield) [reply pvnet/app.py:317]: ah i see now, for the below to do the sun masking. 
Q: do we need to sun mask on national if we have done it on gsp?

File: pvnet/app.py [hunk 4]
```diff
@@ -346,35 +392,95 @@ def app(
...
+        # Multiply normalised forecasts by capacities and clip negatives
+        da_abs_national = da_normed_national.clip(0, None) * national_capacity
+
+        # Apply sundown mask - All GSPs must be masked to mask national
+        da_abs_national = da_abs_national.where(~da_sundown_mask.all(dim="gsp_id")).fillna(0.0)
+
+        da_abs_all = xr.concat([da_abs_national, da_abs], dim="gsp_id")
+
+    else:
+        logger.info("Using simple sum")
```
Comment (peterdudfield) [inline pvnet/app.py:471]: could you expand comment, to make it more clear

**Review by peterdudfield — APPROVED** [review]
Looks good!!! well done. Does it work when you run it locally?

**Comment by dfulu** [comment]
Yeh it runs locally, here's this mornings forecast for comparison
Screenshot 2023-08-15 at 09 53 22

File: pvnet/app.py (hunk 1 above)
Comment (dfulu) [reply pvnet/app.py:252]: I had thought about that, but actually when I ran the line `ds_gsp = next(iter(OpenGSPFromDatabase()))`, `ds_gsp` did not have any information for national - no`gsp_id=0`

File: pvnet/app.py (hunk 3 above)
Comment (dfulu) [reply pvnet/app.py:317]: Yeh I think we do. The summation model will occassionally predict positive values during the night, mainly for the 90th percentile line. So it does need to be masked. Also the summation model was trained on non-sun-masked GSP inputs. To keep things the same I don't mask the GSP inputs fed into the summation model here either

File: pvnet/app.py (hunk 1 above)
Comment (dfulu) [reply pvnet/app.py]: I've added a commit to pull the national capacity from the database

File: pvnet/app.py [hunk 5]
```diff
@@ -398,4 +506,4 @@ def app(
 
 
 if __name__ == "__main__":
-    typer.run(app)
+    2  # typer.run(app)
```
Comment (peterdudfield) [inline pvnet/app.py:509]: This means it doesnt run anymore @dfulu , can you put this back in?>

**Pull request merged**